Generator/doc/Generator/CGAL/point_generators_3.h

```diff
@@ -335,6 +335,213 @@ namespace CGAL {
 
 /*!
 
+The class `Random_points_on_triangle_mesh_3` is an input iterator creating points uniformly
+distributed inside the faces of a triangle mesh model of `FaceListGraph`.
+
+\cgalModels `InputIterator`
+\cgalModels `PointGenerator`
+
+\sa `CGAL::cpp11::copy_n()`
+\sa `CGAL::Counting_iterator`
+\sa `CGAL::Random_points_in_disc_2<Point_2, Creator>`
+\sa `CGAL::Random_points_in_cube_3<Point_3, Creator>`
+\sa `CGAL::Random_points_in_triangle_3<Point_3, Creator>`
+\sa `CGAL::Random_points_on_sphere_3<Point_3, Creator>`
+\sa `CGAL::Random_points_in_triangle_mesh_2<Point_2, Triangulation>`
+\sa `CGAL::Random_points_on_tetrahedral_mesh_boundary<C3t3>`
+\sa `CGAL::Random_points_in_tetrahedral_mesh_3<C3t3>`
+\sa `std::random_shuffle`
+
+*/
+template <class Point_3, class TriangleMesh>
+class Random_points_on_triangle_mesh_3 {
+public:
+
+/// \name Types
+/// @{
+
+/*!
+
+*/
+typedef std::input_iterator_tag iterator_category;
+
+/*!
+
+*/
+typedef Point_3 value_type;
+
+/*!
+
+*/
+typedef std::ptrdiff_t difference_type;
+
+/*!
+
+*/
+typedef const Point_3* pointer;
+
+/*!
+
+*/
+typedef const Point_3& reference;
+
+
+
+/*!
+Creates  an input iterator `g` generating points of type `Point_3` uniformly
+distributed in the mesh faces. Each triangle has a probability to be chosen to hold the point depending on its area.
+
+*/
+Random_points_on_triangle_mesh_3(TriangleMesh& mesh, Random& rnd = default_random);
+
+/// @}
+
+}; /* end Random_points_on_triangle_mesh_3 */
+
+} /* end namespace CGAL */
+
+namespace CGAL {
+
+/*!
+
+The class `Random_points_on_tetrahedral_mesh_boundary` is an input iterator creating points uniformly
+distributed on the boundary of a tetrahedral mesh of type `Mesh_complex_3_in_triangulation_3`.
+
+\cgalModels `InputIterator`
+\cgalModels `PointGenerator`
+
+\sa `CGAL::cpp11::copy_n()`
+\sa `CGAL::Counting_iterator`
+\sa `CGAL::Random_points_in_disc_2<Point_2, Creator>`
+\sa `CGAL::Random_points_in_cube_3<Point_3, Creator>`
+\sa `CGAL::Random_points_in_triangle_3<Point_3, Creator>`
+\sa `CGAL::Random_points_on_sphere_3<Point_3, Creator>`
+\sa `CGAL::Random_points_on_triangle_mesh_3<Point_3, TriangleMesh>`
+\sa `CGAL::Random_points_in_triangle_mesh_2<Point_2, Triangulation>`
+\sa `CGAL::Random_points_in_tetrahedral_mesh_3<C3t3>`
+\sa `std::random_shuffle`
+
+*/
+template <class C3t3>
+class Random_points_on_tetrahedral_mesh_boundary {
+public:
+
+/// \name Types
+/// @{
+
+/*!
+
+*/
+typedef std::input_iterator_tag iterator_category;
+
+/*!
+
+*/
+typedef Weighted_point_3 value_type;
+
```

Comment:
You need to document where `Weighted_point_3` comes from. Same in the other class